### DISCUSSION TITLE
How to handle Docker container failing to build?

### DISCUSSION THREAD

**Discussion by joole** [coder/coder#16261] [General]
+ 1.create a dockerfile template as following 
```
FROM ubuntu

RUN apt-get update \
	&& apt-get install -y \
	curl \
	git \
	golang \
	sudo \
	vim \
	wget \
	bzip2 \
	&& rm -rf /var/lib/apt/lists/*

ARG USER=coder
RUN useradd --groups sudo --no-create-home --shell /bin/bash ${USER} \
	&& echo "${USER} ALL=(ALL) NOPASSWD:ALL" >/etc/sudoers.d/${USER} \
	&& chmod 0440 /etc/sudoers.d/${USER}
USER ${USER}
WORKDIR /home/${USER}

# 下载并安装 Miniconda
RUN wget https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -O /home/${USER}/miniconda.sh \
    && bash /home/${USER}/miniconda.sh -b -p /home/${USER}/miniconda \
    && rm /home/${USER}/miniconda.sh \
    && /home/${USER}/miniconda/bin/conda init bash

# 更新 PATH 变量以包含 conda
ENV PATH="/home/${USER}/miniconda/bin:$PATH"

# 切换到 bash shell
SHELL ["/bin/bash", "-c"]

# 确保 conda 命令可用
RUN conda --version

```

+ 2. create a workspace as upper dockefile
After a prolonged wait, as indicated in the following image, I am unable to cancel the process, and even restarting the entire PC does not resolve the issue.
```
Workspace build failed
send periodic update: context deadline exceeded
```

**Comment by matifali** [comment]
Hi @joole are you able to build this `Doxkerfile` locally with `docker build`?

**Comment by joole** [comment]
@matifali Yes, I has installed 1Panel to manage the server.

**Comment by matifali** [comment]
@joole What do you see in the logs on the coder server?

**Comment by joole** [comment]
@matifali 
![image](https://github.com/coder/coder/assets/2959094/e7969e20-58bd-403a-bb25-d12bbec4c90d) there are the logs output. I can't cancel the staring action even if I restart those relative dockers.

and then I found in the relative workspace's docker's logs show( I started it by myself :
```
2024-05-28 19:23:53.248 [warn]  run exited with error ...
    error= GET https://coder.heycrow.cyou/api/v2/workspaceagents/me/rpc?version=2.0: unexpected status code 401: Workspace agent not authorized.: Try logging in using 'coder login'.
           	Error: The agent cannot authenticate until the workspace provision job has been completed. If the job is no longer running, this agent is invalid.

```

**Comment by matifali** [comment]
Hi @joole it looks like the docker image is building fine but the container is not able to start. Can you share your template to show if `coder_agent`'s `init_script` is correctly being run as the container's enyrypoint.
